rfcs/000-consolidate-amundsen-microservices.md

```diff
@@ -0,0 +1,154 @@
+
+*This change will move all the above into a single repository making it easy to change anything without keeping track of multiple repositories, and redundant code.* 
+
+Having one config file does NOT mean we can not have the incubator integration. It will be the same process as of today i.e., a separate incubation repo, which implements the required methods, and then install that as a requirement and set that as a proxy within the config file.
+
+
+## Transition Path
+Most of the code is already there and is spread across multiple repositories. I can see this happen in multiple phases.
+
+#### Phase I:
+Merge `amundsenmetadatalibrary` and `amundsensearchlibrary`. This should not take too long and will be easy to maintain the same URL patterns for each under separate directories within one repository. 
+The new repository will be named `amundsen`. During this period the `amundsenfrontend` repository will be unchanged and will work as it is. 
+
+Metadata and Search will have their own Flask blueprints. 
+- /api/metadata/
+- /api/search/
+
+During this phase, we will freeze the codebase for `amundsenmetadatalibrary` and `amundsensearchlibrary` repositories for any new features. Security patches will still be approved and merged.
+
+A very high level of how the directories will look like after this phase is below:
+
+<img src="../assets/30/phase1.png" alt="drawing" width="300"/>
+
+#### Phase II:
+In this phase, we’ll deprecate the `amundsenfrontendlibrary` and `amundsencommon` repositories/packages. 
+We’ll move the react application as-is in the `amundsen` repository. 
+Next will be to deprecate the frontend codebase completely and call the metadata and search endpoints directly from the frontend react application. This will completely remove the frontend Flask codebase. 
+All the custom endpoints will be moved to the `amundsen` repository. 
+
+Below is how the folder structure looks like, along with the setup.py file on how to do we set up the dependencies.
+
+<img src="../assets/30/phase2.png" alt="drawing" width="400"/>
+
+<img src="../assets/30/setup.png" alt="drawing" width="500"/>
+
+## How We Communicate This
+
+- We'll promote this RFC frequently on Amundsen Slack and other social media channels, so to get feedback from the community.
+- During the community meetings, update on the progress and phases. 
+- We'll write a Blog/Tutorial about this change on how to migrate your existing Amundsen deployment to this new architecture.
+
+Since this will change the way we deploy and install the Amundsen project, we will not remove any of the existing repositories i.e., amundsenmetadatalibrary, amundsensearchlibrary, amundsenfrontendlibrary.
+We will add a deprecation warning on each readme and packages page where applicable, and introduce the new way i.e., a mono-repo, along the way. All the new work will be done in the new repository. 
+
+## Drawbacks
```

Comment:
Aside from brief development freezes, this migration will probably be at least somewhat painful for folks already using amundsen in the short term (having to migrate custom code over etc). Not to say it isn't worth it, but definitely a tradeoff.